core/local/local.go

```diff
@@ -120,17 +123,28 @@ type Executor struct {
 
 func New(r lib.Runner) *Executor {
 	return &Executor{
-		Runner:   r,
-		Logger:   log.StandardLogger(),
-		endIters: -1,
-		endTime:  -1,
+		Runner:      r,
+		Logger:      log.StandardLogger(),
+		runSetup:    true,
+		runTeardown: true,
+		endIters:    -1,
+		endTime:     -1,
 	}
 }
 
-func (e *Executor) Run(parent context.Context, out chan<- []stats.Sample) error {
+func (e *Executor) Run(parent context.Context, out chan<- []stats.Sample) (reterr error) {
 	e.runLock.Lock()
 	defer e.runLock.Unlock()
 
+	if e.Runner != nil && e.runSetup {
+		setupCtx, setupCancel := context.WithTimeout(parent, 10*time.Second)
```

Comment:
I think that the setup and teardown timeouts should be configurable. Maybe for now we can just put a `TODO` here (and below for teardown) and note it in the [new issue](https://github.com/loadimpact/k6/issues/539) to fix later, but just yesterday I chatted with a user that had to wait for an SMS message to be received in their registration step (that they'll probably want to use `setup()` for), for which 10 seconds woudn't be nearly enough.

Reply:
Agree, the timeouts should be configurable.